DuranVillanuevaJean/practica3/p3.py

```diff
@@ -0,0 +1,257 @@
+    [['bi', '3.cpl'], ['t', 'lig'], ["'u̱n", 'stem'], ['gí', '1.obj'], 'v'],          #palabra
+    [['bi', '3.cpl'], ['mähtratá', 'stem'], ['gí', '1.obj'], 'v'],                   #palabra
+    [['ko', 'stem'], 'obl'],                                                         #palabra
+    [['chíkóhté', 'stem'], 'obl']                                                    #palabra
+  ]
+  """
+  """
+    [['n', 'psd'], ['dó', '1.cpl'], ['phu̱di', 'stem'], 'v']  ----- la palabra es -----> n-dó-phu̱di
+                                                            ----- su etiqueta es ----->     v
+
+      ['n', 'psd'], ['dó', '1.cpl'], ['phu̱di', 'stem']      <----- morfologia de la palabra -------
+  """
+
+
+  """## Feature functions: corpus"""
+
+  """
+  Feature function only take the phrase, where the phrase has words and its tags
+  """
+  def feature(corpus):
+    cps = []
+    for phrase in corpus:
+      cps.append(phrase_to_corpus_elmt(phrase))
+    return cps
+
+  """
+  Take a phrase and converts all of its elements to just the words and its tags
+
+  [['n', 'psd'], ['dó', '1.cpl'], ['phu̱di', 'stem'], 'v']         -> ["ndóphu̱di", "v"]
+  [['bi', '3.cpl'], ['mähtratá', 'stem'], ['gí', '1.obj'], 'v']   -> ["bimähtratágí", v]
+  """
+  def phrase_to_corpus_elmt(phrase):
+    lst = []
+    for w_n_t in phrase:
+      tag = w_n_t[-1] #tag
+      raw_word = w_n_t[:-1]
+      word = aux(raw_word)
+      #print([word,tag])
+      lst.append([word,tag])
+    return lst
+
+  """
+  Take the parts of a word  and contats them
+
+  [['n', 'psd'], ['dó', '1.cpl'], ['phu̱di', 'stem']] -> ["ndóphu̱di"]
+  """
+  def aux(raw_word):
+    word = ""
+    for part in raw_word:
+      word += part[0]
+    return word
+
+  corpus = feature(raw_corpus)
+
+
+  """## Feature functions 2: X, y
+
+  Dividimos a los x de los y
+  """
+
+  def divide(corpus):
+    X = []
+    Y = []
+    for phrase in corpus:
+      x, y = divide_x_n_y(phrase)
+      X.append(x)
+      Y.append(y)
+    return X, Y
+
+  def divide_x_n_y(phrase):
+    X = []
+    Y = []
+    for x_n_y in phrase:
+      x = x_n_y[0]
+      y = x_n_y[1]
+      X.append(x)
+      Y.append(y)
+    return X, Y
+
+
+  """#### Antes que nada, volvemos a codificar cada cadena de X"""
+
+  X, Y = divide(corpus[:5])
+
+
+  """Sí funcionó, ahora hagamoslo con todo el corpus:"""
+
+  #pip install unidecode
+
+  from unidecode import unidecode
+
+  X, Y = divide(corpus)
+
+  Y_1 = []
+  for y in Y:
+    phrase = []
+    for word in y:
+      word = unidecode(word)
+      phrase.append(word)
+    Y_1.append(phrase)
+
+  y = Y_1
+
+  """## Feature functions 3: final_corpus"""
+
+  def word_to_features(sent, i):
+      word = sent[i][0]
+      features = {
+          'word.lower()': word.lower(),
+          'word[:3]': word[:3],
+          'word[:2]': word[:2],
+          'word.isupper()': word.isupper(),
+          'word.istitle()': word.istitle(),
+          'word.isdigit()': word.isdigit(),
```

Comment:
Estas features parecen ser inútiles, ya que el corpus está en minúsculas y no hay números